Not building on CPU without avx512

```
[sin3point14@lolikawa build]$ cmake --version
cmake version 4.0.2-dirty

CMake suite maintained and supported by Kitware (kitware.com/cmake).
[sin3point14@lolikawa build]$ g++ --version
g++ (GCC) 15.1.1 20250425
Copyright (C) 2025 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

[sin3point14@lolikawa build]$ make -j6
[  0%] Built target CLI11
[ 50%] Built target spdlog
[ 75%] Building CXX object CMakeFiles/tight_inclusion.dir/src/tight_inclusion/interval_root_finder.cpp.o
[ 75%] Building CXX object CMakeFiles/tight_inclusion.dir/src/tight_inclusion/logger.cpp.o
[ 75%] Building CXX object CMakeFiles/tight_inclusion.dir/src/tight_inclusion/avx.cpp.o
[ 75%] Building CXX object CMakeFiles/tight_inclusion.dir/src/tight_inclusion/interval.cpp.o
[ 81%] Building CXX object CMakeFiles/tight_inclusion.dir/src/tight_inclusion/ccd.cpp.o
c++: error: unrecognized command-line option ‘-mno-avx512pf’; did you mean ‘-mno-avx512f’?
c++: error: unrecognized command-line option ‘-mno-avx512pf’; did you mean ‘-mno-avx512f’?
c++: error: unrecognized command-line option ‘-mno-avx512pf’; did you mean ‘-mno-avx512f’?
c++: error: unrecognized command-line option ‘-mno-avx512pf’; did you mean ‘-mno-avx512f’?
c++: error: unrecognized command-line option ‘-mno-avx512pf’; did you mean ‘-mno-avx512f’?
c++: error: unrecognized command-line option ‘-mno-avx512er’; did you mean ‘-mno-avx512bw’?
c++: error: unrecognized command-line option ‘-mno-avx512er’; did you mean ‘-mno-avx512bw’?
c++: error: unrecognized command-line option ‘-mno-avx512er’; did you mean ‘-mno-avx512bw’?
c++: error: unrecognized command-line option ‘-mno-avx512er’; did you mean ‘-mno-avx512bw’?
c++: error: unrecognized command-line option ‘-mno-avx512er’; did you mean ‘-mno-avx512bw’?
c++: error: unrecognized command-line option ‘-mno-avx512pf’; did you mean ‘-mno-avx512f’?
c++: error: unrecognized command-line option ‘-mno-avx512pf’; did you mean ‘-mno-avx512f’?
c++: error: unrecognized command-line option ‘-mno-avx512pf’; did you mean ‘-mno-avx512f’?
c++: error: unrecognized command-line option ‘-mno-avx512pf’; did you mean ‘-mno-avx512f’?
c++: error: unrecognized command-line option ‘-mno-avx512pf’; did you mean ‘-mno-avx512f’?
c++: error: unrecognized command-line option ‘-mno-avx512er’; did you mean ‘-mno-avx512bw’?
c++: error: unrecognized command-line option ‘-mno-avx512er’; did you mean ‘-mno-avx512bw’?
c++: error: unrecognized command-line option ‘-mno-avx512er’; did you mean ‘-mno-avx512bw’?
make[2]: *** [CMakeFiles/tight_inclusion.dir/build.make:79: CMakeFiles/tight_inclusion.dir/src/tight_inclusion/avx.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[2]: *** [CMakeFiles/tight_inclusion.dir/build.make:107: CMakeFiles/tight_inclusion.dir/src/tight_inclusion/interval_root_finder.cpp.o] Error 1
make[2]: *** [CMakeFiles/tight_inclusion.dir/build.make:135: CMakeFiles/tight_inclusion.dir/src/tight_inclusion/logger.cpp.o] Error 1
c++: error: unrecognized command-line option ‘-mno-avx512er’; did you mean ‘-mno-avx512bw’?
make[2]: *** [CMakeFiles/tight_inclusion.dir/build.make:93: CMakeFiles/tight_inclusion.dir/src/tight_inclusion/ccd.cpp.o] Error 1
c++: error: unrecognized command-line option ‘-mno-avx512er’; did you mean ‘-mno-avx512bw’?
make[2]: *** [CMakeFiles/tight_inclusion.dir/build.make:121: CMakeFiles/tight_inclusion.dir/src/tight_inclusion/interval.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:191: CMakeFiles/tight_inclusion.dir/all] Error 2
make: *** [Makefile:156: all] Error 2
```

--- a/cmake/find/FindAVX.cmake
+++ b/cmake/find/FindAVX.cmake
@@ -116,8 +116,14 @@ if(CMAKE_COMPILER_IS_GNUCC OR CMAKE_COMPILER_IS_GNUCXX OR CMAKE_CXX_COMPILER_ID 
   elseif(DETECTED_AVX_10)
       SET(AVX_FLAGS "${AVX_FLAGS} -mavx")
   endif()
-  # TODO: Check for AVX512 support
-  SET(AVX_FLAGS "${AVX_FLAGS} -mno-avx512f -mno-avx512pf -mno-avx512er -mno-avx512cd")
+  # Only add -mno-avx512* if the compiler accepts them
+  foreach(flag -mno-avx512f -mno-avx512pf -mno-avx512er -mno-avx512cd)
+    string(REPLACE "-" "_" safe_flag ${flag})
+    check_cxx_compiler_flag("${flag}" HAS_FLAG_${safe_flag})
+    if(HAS_FLAG_${safe_flag})
+      set(AVX_FLAGS "${AVX_FLAGS} ${flag}")
+    endif()
+  endforeach()
 elseif(CMAKE_CXX_COMPILER_ID MATCHES "Intel")
   set(AVX_FLAGS "-xHost")
 elseif(MSVC)

